# Subir archivos

In [ ]:
from google.colab import files
uploaded = files.upload()


Saving feo.csv to feo (1).csv


In [ ]:
import pandas as pd


In [ ]:
df= pd.read_csv('feo.csv', encoding='latin-1')

In [ ]:
df

tamaño hielo      sabor  score
0  grande    sin    kahlua    8.2
1  grande    con    kahlua    9.0
2   chica    sin    kahlua    7.0
3   chica    con    kahlua    7.5
4  grande    con  horchata   10.0
5   chica    sin  horchata    6.0
6   chica    con  horchata    6.4
7  grande    sin  horchata    9.5

In [ ]:
#convertir hielos a booleanos

df['hielo '] =(df['hielo '] == "con").astype(int)
df['tamaño'] = (df['tamaño'] == 'grande').astype(int)
df['kahlua']= (df['sabor'] == 'kahlua').astype(int)
df['horchata']= (df['sabor'] == 'horchata').astype(int)
df.drop('sabor', axis=1, inplace=True)
df


tamaño  hielo   score  kahlua  horchata
0       1       0    8.2       1         0
1       1       1    9.0       1         0
2       0       0    7.0       1         0
3       0       1    7.5       1         0
4       1       1   10.0       0         1
5       0       0    6.0       0         1
6       0       1    6.4       0         1
7       1       0    9.5       0         1

In [ ]:
import numpy as np

In [ ]:
tamaño= df['tamaño'].to_numpy().reshape(-1,1)
hielo =df['hielo '].to_numpy().reshape(-1,1)
horchata=df['horchata'].to_numpy().reshape(-1,1)
kahlua= df['kahlua'].to_numpy().reshape(-1,1)
X= np.hstack((tamaño, hielo, kahlua, horchata))
y= df['score'].to_numpy().reshape(-1,1)

In [ ]:
import statsmodels.api as sm

In [ ]:
X = sm.add_constant(X)
model=sm.OLS(y, X) # -> Modela minimos cuadrados ordinarios con todas las variables
results=model.fit() #-> Optimiza
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Mon, 09 Feb 2026   Prob (F-statistic):             0.0473
Time:                        23:49:05   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.0

# Parte 2

In [ ]:
df['hielo '] = df['hielo '].replace(0, -1)
df['tamaño'] = df['tamaño'].replace(0,-1)
df

tamaño  hielo   score  kahlua  horchata
0       1      -1    8.2       1         0
1       1       1    9.0       1         0
2      -1      -1    7.0       1         0
3      -1       1    7.5       1         0
4       1       1   10.0       0         1
5      -1      -1    6.0       0         1
6      -1       1    6.4       0         1
7       1      -1    9.5       0         1

In [ ]:
tamaño= df['tamaño'].to_numpy().reshape(-1,1)
hielo =df['hielo '].to_numpy().reshape(-1,1)
horchata=df['horchata'].to_numpy().reshape(-1,1)
kahlua= df['kahlua'].to_numpy().reshape(-1,1)
X= np.hstack((tamaño, hielo, kahlua, horchata))
y= df['score'].to_numpy().reshape(-1,1)

In [ ]:
X = sm.add_constant(X)
model=sm.OLS(y, X) # -> Modela minimos cuadrados ordinarios con todas las variables
results=model.fit() #-> Optimiza
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Tue, 10 Feb 2026   Prob (F-statistic):             0.0473
Time:                        00:12:39   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.3000      0.185     28.644      0.0

# Parte 3
Comparación bethas y observaciones


El intercepto nos ayuda a ver que el salon cree que las aguas del feo no pueden ser peor que un 5.3, podemos observar también que nos importa mucho más el sabor de las aguas que el tamaño o los hielos, pero ambos sabores nos gustan igual, podemos observar que prácticamente no nos importa si el agua tiene hielo, aunque si alcanzamos a penalizar un poco por el tamaño.